public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -147,17 +161,39 @@ export class AppChromeService {
 
     const lastEntry = entries[0];
     const newEntry = { name: newPageNav.text, views: [], breadcrumbs, time: Date.now(), url: window.location.href };
-    const isSameUrl = lastEntry && newEntry.url === lastEntry.url;
 
-    // To avoid adding an entry with the same url twice, we always use the latest one
-    if (isSameUrl) {
-      entries[0] = newEntry;
+    const getEntryPath = (entry: HistoryEntry) =>
+      entry.url.substring(0, entry.url.indexOf('?') !== -1 ? entry.url.indexOf('?') : undefined);
+    const getEntryQueryParams = (entry: HistoryEntry) =>
+      entry.url.indexOf('?') !== -1 ? entry.url.substring(entry.url.indexOf('?')) : undefined;
+
+    const newEntryPath = getEntryPath(newEntry);
+    const newEntryQueryParams = getEntryQueryParams(newEntry);
+    const lastEntryPath = lastEntry && getEntryPath(lastEntry);
+    const lastEntryQueryParams = lastEntry && getEntryQueryParams(lastEntry);
+    const isSamePath = newEntryPath === lastEntryPath;
+    const isSameQueryParams = newEntryQueryParams === lastEntryQueryParams;
+    // const isSameURL = isSamePath && isSameQueryParams;
+    console.log('params', newEntryQueryParams);
+    console.log('test', (newEntry.url.includes('\/explore') || newEntry.url.includes('\/d\/')) && isSameQueryParams);
+    if (isSamePath) {
+      if ((newEntry.url.includes('\/explore?') || newEntry.url.includes('\/d\/')) && isSameQueryParams) {
+        entries[0] = newEntry;
+      } else if (lastEntry.url.includes('\/d\/') && !lastEntryQueryParams) {
+        entries[0] = newEntry;
+      } else {
+        entries = [newEntry, ...entries];
+      }
```

Comment:
can we move this logic to a history service and add tests because this looks very messy

Reply:
Hi @torkelo,
Yes 😅 I had that already in mind but did not move the code till now.